etcdserver/server_test.go

```diff
@@ -698,6 +700,46 @@ func TestSyncTrigger(t *testing.T) {
 	}
 }
 
+func TestCreateRaftSnapshot(t *testing.T) {
+	s := newRaftStorage()
+	s.Append([]raftpb.Entry{{Index: 1, Term: 1}})
+	st := &storeRecorder{}
+	srv := &EtcdServer{
+		r: raftNode{
+			raftStorage: s,
+		},
+		store: st,
+	}
+	snap := srv.createRaftSnapshot(1, raftpb.ConfState{Nodes: []uint64{1}})
+
+	gaction := st.Action()
+	if len(gaction) != 2 {
+		t.Fatalf("len(action) = %d, want 2", len(gaction))
+	}
+	if !reflect.DeepEqual(gaction[0], testutil.Action{Name: "Clone"}) {
+		t.Errorf("action = %s, want Clone", gaction[0])
+	}
+	if !reflect.DeepEqual(gaction[1], testutil.Action{Name: "SaveNoCopy"}) {
+		t.Errorf("action = %s, want SaveNoCopy", gaction[1])
+	}
+
+	wdata, err := st.Save()
```

Comment:
let's move the preparation phase before the actual testing phase to line 712? it would improve readability.